test/e2e/case13_templatization_test.go

```diff
@@ -483,4 +461,90 @@ var _ = Describe("Test templatization", Ordered, func() {
+		)
+
+		It("Should enforce the labels on the e2e-object-variable ConfigMaps", func(ctx SpecContext) {
+			By("Creating the prerequisites")
+			utils.Kubectl("apply", "-f", preReqs)
+
+			By("Applying the " + policyName + " ConfigurationPolicy")
+			utils.Kubectl("apply", "-n", testNamespace, "-f", policyYAML)
+
+			By("By verifying that the ConfigurationPolicy is compliant and has the correct related objects")
+			Eventually(func(g Gomega) {
+				managedPlc := utils.GetWithTimeout(
+					clientManagedDynamic, gvrConfigPolicy, policyName, testNamespace, true, defaultTimeoutSeconds,
+				)
+
+				utils.CheckComplianceStatus(g, managedPlc, "Compliant")
+
+				relatedObjects, _, _ := unstructured.NestedSlice(managedPlc.Object, "status", "relatedObjects")
+				g.Expect(relatedObjects).To(HaveLen(2))
+
+				realtedObject1, ok := relatedObjects[0].(map[string]interface{})
+				g.Expect(ok).To(BeTrue(), "Related object was not a map")
+
+				relatedObject1NS, _, _ := unstructured.NestedString(realtedObject1, "object", "metadata", "namespace")
+				g.Expect(relatedObject1NS).To(Equal("case13-e2e-object-variables"))
+
+				realtedObject2, ok := relatedObjects[1].(map[string]interface{})
+				g.Expect(ok).To(BeTrue(), "Related object was not a map")
+
+				relatedObject2NS, _, _ := unstructured.NestedString(realtedObject2, "object", "metadata", "namespace")
+				g.Expect(relatedObject2NS).To(Equal("default"))
+			}, defaultTimeoutSeconds, 1).Should(Succeed())
+
+			By("By verifying the ConfigMaps")
+			configMap1, err := clientManaged.CoreV1().ConfigMaps("case13-e2e-object-variables").Get(
+				ctx, "case13-e2e-object-variables", metav1.GetOptions{},
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(configMap1.ObjectMeta.Labels["case13"]).To(Equal("passed"))
+			Expect(configMap1.ObjectMeta.Labels["namespace"]).To(Equal("case13-e2e-object-variables"))
+
+			configMap2, err := clientManaged.CoreV1().ConfigMaps("default").Get(
+				ctx, "case13-e2e-object-variables", metav1.GetOptions{},
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(configMap2.ObjectMeta.Labels["case13"]).To(Equal("passed"))
+			Expect(configMap2.ObjectMeta.Labels["namespace"]).To(Equal("default"))
+		})
+
+		It("Should fail when the namespace doesn't match after template resolution", func(ctx SpecContext) {
+			By("Applying the " + policyName + " ConfigurationPolicy")
+			utils.Kubectl("apply", "-n", testNamespace, "-f", invalidPolicyYAML)
+
+			By("By verifying that the ConfigurationPolicy is noncompliant")
+			Eventually(func(g Gomega) {
+				managedPlc := utils.GetWithTimeout(
+					clientManagedDynamic,
+					gvrConfigPolicy,
+					invalidPolicyName,
+					testNamespace,
+					true,
+					defaultTimeoutSeconds,
+				)
+
+				utils.CheckComplianceStatus(g, managedPlc, "NonCompliant")
+				g.Expect(utils.GetStatusMessage(managedPlc)).To(Equal(
+					"The object definition's namespace must match the selected namespace after template resolution",
```

Comment:
I think this is a nice approach - I haven't looked at the implementation yet, but is it still possible to "hard-code" a namespace into the object? Or do you have to use `{{.ObjectNamespace}}` when using a namespace selector now?

Using this test's scenario, would it be invalid to set the namespace on the object to be `default`? Because the namespace selector should select two namespaces?

Reply:
@JustinKuli it's a bit confusing but if the user specifies a namespace explicitly, the namespace selector is ignored. If the user has an empty namespace or leverages the `{{ .ObjectNamespace }}` template variable, then the namespace selector is leveraged. If after template resolution the namespace on the object definition is not empty or the current namespace in the namespace selector, it is considered invalid.

This was brought up by @dhaiducek since without this restriction, you could create an object in another namespace than the one that was selected.